ipa-core/src/protocol/ipa_prf/shuffle/sharded.rs

```diff
@@ -441,14 +480,25 @@ where
             )
             .map_ok(|((), c1)| S::new(c1 + c2, a))
         }))
-        .await?)
+        .await?;
+
+    Ok((
+        res,
+        IntermediateShuffleMessages {
+            x1_or_y1: Some(y1),
+            x2_or_y2: Some(y2),
+        },
+    ))
 }
 
 /// Entry point to execute sharded shuffle.
 /// ## Errors
 /// Failure to communicate over the network, either to other MPC helpers, and/or to other shards
 /// will generate a shuffle error.
-pub async fn shuffle<I, S, C>(ctx: C, shares: I) -> Result<Vec<S>, crate::error::Error>
+pub async fn shuffle<I, S, C>(
```

Comment:
nit: I would probably note here that this function must not be called directly (maybe we should even hide it)